src/actions/AppActions.ts

```diff
@@ -108,3 +121,76 @@ export function logLn(message: string, kind: "" | "info" | "warn" | "error" = ""
     kind,
   } as LogLnAction);
 }
+
+export interface SplitGroupAction extends AppAction {
+  type: AppActionType.SPLIT_GROUP;
+}
+
+export function splitGroup() {
+  dispatcher.dispatch({
+    type: AppActionType.SPLIT_GROUP
+  } as SplitGroupAction);
+}
+
+export interface OpenFileAction extends AppAction {
+  type: AppActionType.OPEN_FILE;
+  file: File;
+  preview: boolean;
+}
+
+export function openFile(file: File, preview = true) {
+  dispatcher.dispatch({
+    type: AppActionType.OPEN_FILE,
+    file,
+    preview
+  } as OpenFileAction);
+}
+
+export interface CloseFileAction extends AppAction {
+  type: AppActionType.CLOSE_FILE;
+  file: File;
+}
+
+export function closeFile(file: File) {
+  dispatcher.dispatch({
+    type: AppActionType.CLOSE_FILE,
+    file
+  } as CloseFileAction);
+}
+
+export interface OpenProjectFilesAction extends AppAction {
+  type: AppActionType.OPEN_PROJECT_FILES;
+  json: ProjectTemplate;
+}
+
+export function openProjectFiles(json: Object) {
```

Comment:
make this function async and move:

```
const projectModel = new Project();
const json = await Service.loadProject(project, projectModel);
```
and dispatch `projectModel` and `json.openedFiles`. Notice that ideally we need to create a new project. Similar to load from fiddle (can we use LOAD_PROJECT action for that?)